odinmain/src/main/kotlin/me/odinmain/features/impl/floor7/DragonCheck.kt

```diff
@@ -14,55 +11,36 @@ import me.odinmain.features.impl.floor7.WitherDragons.sendTime
 import me.odinmain.features.impl.skyblock.ArrowHit.onDragonSpawn
 import me.odinmain.features.impl.skyblock.ArrowHit.resetOnDragons
 import me.odinmain.utils.isVecInXZ
+import me.odinmain.utils.skyblock.PlayerUtils
 import me.odinmain.utils.skyblock.modMessage
 import net.minecraft.entity.boss.EntityDragon
 import net.minecraft.entity.item.EntityArmorStand
 import net.minecraft.init.Blocks
 import net.minecraft.item.Item
 import net.minecraft.network.play.server.S04PacketEntityEquipment
+import net.minecraft.network.play.server.S0FPacketSpawnMob
+import net.minecraft.network.play.server.S1CPacketEntityMetadata
+import net.minecraft.util.Vec3
 import net.minecraftforge.event.entity.EntityJoinWorldEvent
-import net.minecraftforge.event.entity.living.LivingDeathEvent
 
 object DragonCheck {
 
     var lastDragonDeath: WitherDragonsEnum = WitherDragonsEnum.None
-    var dragonEntityList = emptyList<EntityDragon>()
+    val dragonEntityList = mutableListOf<EntityDragon>()
 
-    fun dragonJoinWorld(event: EntityJoinWorldEvent) {
-        val entity = event.entity as? EntityDragon ?: return
-        val dragon = WitherDragonsEnum.entries.find { isVecInXZ(entity.positionVector, it.boxesDimensions) } ?: return
-        if (dragon.state == WitherDragonState.DEAD) return
-
-        dragon.state = WitherDragonState.ALIVE
-        dragon.timeToSpawn = 100
-        dragon.timesSpawned += 1
-        dragon.entity = entity
-        dragon.spawnedTime = System.currentTimeMillis()
-        dragon.isSprayed = false
-
-        if (sendArrowHit && WitherDragons.enabled) arrowSpawn(dragon)
-        if (resetOnDragons && WitherDragons.enabled) onDragonSpawn()
-        if (sendSpawned && WitherDragons.enabled) {
-            val numberSuffix = when (dragon.timesSpawned) {
-                1 -> "st"
-                2 -> "nd"
-                3 -> "rd"
-                else -> "th"
-            }
-            modMessage("§${dragon.colorCode}${dragon.name} §fdragon spawned. This is the §${dragon.colorCode}${dragon.timesSpawned}${numberSuffix}§f time it has spawned.")
-        }
+    fun dragonUpdate(packet: S1CPacketEntityMetadata) {
+        val dragon = WitherDragonsEnum.entries.find { it.entityId == packet.entityId } ?: return
+        if (dragon.entity == null) return dragon.updateEntity(packet.entityId)
+        val health = packet.func_149376_c().find { it.dataValueId == 6 }?.`object` as? Float ?: return
+        if (health > 0 || dragon.state == WitherDragonState.DEAD) return
+        dragon.setDead()
```

Comment:
Since there is only one loc after the if statement, it should be inverted instead of it returning.